ballerina/tests/csv_io.bal

```diff
@@ -99,6 +120,139 @@ isolated function testReadCsv() returns Error? {
+    stream<Employee4, Error?> result = check fileReadCsvAsStream(filePath); 
+    int i = 0;
+    check result.forEach(function(Employee4 val) {
+        test:assertEquals(val.salary, expected[i]);
+        i = i + 1;
+    });
+    test:assertEquals(i, 3);
+}
+
+@test:Config {}
+function testReadFileCsvAsStreamUsingResourceFileOpenRecord() returns error? {
+    string filePath = RESOURCES_BASE_PATH + "datafiles/io/records/sample5b.csv";
+    int[] expected = [10000,20000,30000];
+    stream<record {string id;string name;int salary;}, Error?> result = check fileReadCsvAsStream(filePath);
+    int i = 0;
+    check result.forEach(function(record {string id; string name; int salary;} val) {
+        test:assertEquals(val.salary, expected[i]);
+        i = i + 1;
+    });
+    test:assertEquals(i, 3);
+}
+
+@test:Config {}
+isolated function testWriteRecordCsv() returns Error? {
+    Employee E = {
+        id: "1",
+        name: "Foo1",
+        salary: 100000.0f
+    };
+    Employee B = {
+        id: "2",
+        name: "Foo2",
+        salary: 300000.0f
+    };
+    string filePath = TEMP_DIR + "recordsDefault_records.csv";
+    Employee[] content1 = [E, B];
+    test:assertEquals(check fileWriteCsv(filePath, content1), ());
+}
+
+@test:Config {dependsOn: [testWriteRecordCsv]}
+function testWriteRecordCsvRead() returns error? {
+    Employee E = {
+        id: "1",
+        name: "Foo1",
+        salary: 100000.0f
+    };
+    Employee B = {
+        id: "2",
+        name: "Foo2",
+        salary: 300000.0f
+    };
+    Employee[] expected = [E, B];
+    string filePath = TEMP_DIR + "recordsDefault_records.csv";
+    stream<Employee, Error?> result = check fileReadCsvAsStream(filePath);
+    int i = 0;
+    check result.forEach(function(Employee val) {
+        test:assertEquals(val.salary, expected[i].salary);
+        i = i + 1;
+    });
+    test:assertEquals(i, 2);
+}
+
+@test:Config {}
+isolated function testWriteRecordCsvAsStream() returns Error? {
+    Employee E = {
+        id: "1",
+        name: "Foo1",
+        salary: 100000.0f
+    };
+    Employee B = {
+        id: "2",
+        name: "Foo2",
+        salary: 300000.0f
+    };
+
+    string filePath = TEMP_DIR + "recordsDefault_stream.csv";
+    Employee[] content1 = [E, B];
+    test:assertEquals(check fileWriteCsvFromStream(filePath, content1.toStream()), ());
+}
+
+@test:Config {dependsOn: [testWriteRecordCsvAsStream]}
+function testWriteRecordCsvAsStreamRead() returns error? {
+    Employee E = {
+        id: "1",
+        name: "Foo1",
+        salary: 100000.0f
+    };
+    Employee B = {
+        id: "2",
+        name: "Foo2",
+        salary: 300000.0f
+    };
+    Employee[] expected = [E, B];
+    string filePath = TEMP_DIR + "recordsDefault_stream.csv";
+    stream<Employee, Error?> result = check fileReadCsvAsStream(filePath);
+    int i = 0;
+    check result.forEach(function(Employee val) {
+        test:assertEquals(val.salary, expected[i].salary);
```

Comment:
Better to assert other fields as well. Shall we simply assert `expected` and `result`? 
Check other places as well.